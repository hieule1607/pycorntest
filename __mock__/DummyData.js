// User Favorited Data
const favoriteUser = {
    "seed": "d9f52b82d43934c7",
    "user": {
        "SSN": "846-92-6871",
        "cell": "(300)-241-9731",
        "dob": "106070258",
        "email": "dave.montgomery50@example.com",
        "gender": "male",
        "location": {
            "city": "missoula",
            "state": "maryland",
            "street": "6057 n dapple gray rdpark st",
            "zip": "32703",
        },
        "md5": "1fe7ab70db63f0fa14683aa4d5dfd58f",
        "name": {
            "first": "dave",
            "last": "montgomery",
            "title": "mr",
        },
        "password": "samson",
        "phone": "(373)-596-5326",
        "picture": "http://api.randomuser.me/portraits/men/92.jpg",
        "registered": "1509649695",
        "salt": "CvUOdHWG",
        "sha1": "0e5fdd5571c55f722d3fcf691b4c9e0e12defe17",
        "sha256": "0c8366098bdae28b04725594cd06d03868ad7a721eb92d2f7b81881e6036f45a",
        "username": "greensnake91",
    },
    "version": "0.4",
  }

  // The Response is returned from API
  const responseData = {"results": [{
    "seed": "977d45b116b8ae36",
    "user": {
        "SSN": "210-96-3772",
        "cell": "(245)-107-1291",
        "dob": "283069081",
        "email": "claude.walker32@example.com",
        "gender": "male",
        "location": {
            "city": "grand prairie",
            "state": "new york",
            "street": "4881 sunset st",
            "zip": "36711",
        },
        "md5": "39c15c1125c6b715ef97b336c069c5da",
        "name": {
            "first": "claude",
            "last": "walker",
            "title": "mr",
        },
        "password": "dixie",
        "phone": "(667)-392-5609",
        "picture": "http://api.randomuser.me/portraits/men/40.jpg",
        "registered": "1465681460",
        "salt": "prjX5a6C",
        "sha1": "5b48fcb100cc3bbb1c1ae4df4680a4b724d299c0",
        "sha256": "fd48fe9999fa19bc403bda3dfb888732b0b350e9dd08511a2e6d94346500b8c6",
        "username": "bigmouse60",
    },
    "version": "0.4",
}]} 

export {favoriteUser, responseData}