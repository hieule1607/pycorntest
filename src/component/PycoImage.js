import React from 'react';
import { View, StyleSheet, Image } from 'react-native';

const CustomImage = (props) => {

  const { uri , styleView } = props;

  return (
    <View style={[ styleView, styles.avatarConner, {borderRadius: styleView.width}]}>
        <Image
        style={[styles.avatarImag, {borderRadius: (styleView.width * 0.9 / 2)}]}
        source={{uri : uri}} 
        imageStyle={{resizeMode: 'cover'}}/>
    </View>
  );
};

const styles = StyleSheet.create({
    avatarConner: {
        flex: 1,
        alignSelf: 'center',
        alignItems: "center",
        justifyContent: "center",
        backgroundColor: "white",
        borderWidth: 1,
        borderColor: '#d6d7da',
        },

    avatarImag: {
        width: '90%',
        height: '90%',       
        },
})

export default CustomImage