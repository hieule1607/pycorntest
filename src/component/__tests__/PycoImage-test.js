import React from 'react';
import renderer from 'react-test-renderer';
import PycoImage from '../PycoImage';
import {favoriteUser} from '../../../__mock__/DummyData';

describe(`PycoImage Component`, () => {
    let props
    let styleView
    beforeEach(()=>{
        styleView = {
            width: 110, 
            height: 110, 
            marginTop: -70
          }
        props = {
            uri:  favoriteUser.user.picture,
            styleView: styleView
        }
    })

    it(`should render correctly`, () => {
        const tree = renderer.create(<PycoImage {...props}/>).toJSON()
        expect(tree).toMatchSnapshot();
    })

})