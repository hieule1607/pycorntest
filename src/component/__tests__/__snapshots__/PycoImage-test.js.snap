// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PycoImage Component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 110,
        "marginTop": -70,
        "width": 110,
      },
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderColor": "#d6d7da",
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "borderRadius": 110,
      },
    ]
  }
>
  <Image
    imageStyle={
      Object {
        "resizeMode": "cover",
      }
    }
    source={
      Object {
        "uri": "http://api.randomuser.me/portraits/men/92.jpg",
      }
    }
    style={
      Array [
        Object {
          "height": "90%",
          "width": "90%",
        },
        Object {
          "borderRadius": 49.5,
        },
      ]
    }
  />
</View>
`;
